neuro_san/test/chat_mock_llm/chat_mock_llm.py

```diff
@@ -0,0 +1,194 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import AIMessage
+from langchain_core.messages import AIMessageChunk
+from langchain_core.messages import BaseMessage
+from langchain_core.outputs import ChatGeneration
+from langchain_core.outputs import ChatGenerationChunk
+from langchain_core.outputs import ChatResult
+from pydantic import ConfigDict
+from pydantic import Field
+from tiktoken import get_encoding
+
+
+class ChatMockLlm(BaseChatModel):
+    """
+    A custom chat model that echoes the input.
+
+    Adapted from https://python.langchain.com/docs/how_to/custom_chat_model/
+    """
+
+    # This is required field and it is possible to have multiple test models.
+    model_name: str = Field(default=None, alias="model")
+    # Maybe useful for testing
+    max_retries: Optional[int] = None
+
+    # Accept both argument name and alias
+    model_config = ConfigDict(populate_by_name=True)
+
+    def _generate(
```

Comment:
Would it be better to use an asynchronous version of this like _agenerate() given our async environment?